### PR TITLE
Fix prysm tests

### PR DESCRIPTION
**Pull request by 0x00101010**
- Updated all clients to latest commit (except for prysm, prysm is now at https://github.com/prysmaticlabs/prysm/commit/db6474a3, which is very close to tip)
- Added an easier to configure way to generate time based genesis for both EL / CL
- Get prysm tests to work

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -4,7 +4,7 @@
 	branch = eip-4844
 [submodule "prysm/prysm"]
 	path = prysm/prysm
-	url = https://github.com/Inphi/prysm
+	url = https://github.com/prysmaticlabs/prysm
 	branch = eip-4844
```
Comment (Inphi) [inline .gitmodules]: the upstream branch is `eip4844`.

Comment (0x00101010) [reply .gitmodules]: Updated, I pulled the latest change from that branch though

File: shared/genesis.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        "berlinBlock": 0,
+        "londonBlock": 0,
+        "shanghaiTime": SHANGHAI_TIME,
+        "shardingForkTime": SHARDING_FORK_TIME,
+        "clique": {
+            "period": 5,
+            "epoch": 30000
+        },
+        "terminalTotalDifficulty": 2,
+        "terminalTotalDifficultyPassed": true
```
Comment (Inphi) [inline shared/genesis.json]: this should be `false` unless you're starting from post-merge genesis.

File: shared/setup.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#!/bin/sh
+
+set -e
+
+# setup shared env vars
+. ./shared.env
+
+GENESIS=$(($(date +%s) + 60))   # 60s till genesis
+SHANGHAI=$(($GENESIS + 108))    # 108s till shanghai
+CANCUN=$(($GENESIS + 144))      # 144s till cancun
```
Comment (Inphi) [inline shared/setup.sh:13]: These values should correspond to the `CAPELLA_FORK_EPOCH` and `EIP4844_FORK_EPOCH` in the chain config. So they should be 128 for capella ( $CapellaForkEpoch \times SecondsPerSlot \times SlotsPerEpoch$ ) and 256 for EIP-4844.

File: shared/genesis.json (hunk 2 above)
Comment (Inphi) [reply shared/genesis.json]: I'm assuming you are generating a post-merge genesis given the bellatrix fork epoch in the chain config.

File: prysm/Dockerfile.prysm [hunk 4]
```diff
@@ -1,28 +1,24 @@
...
-RUN go build -o /build/bootnode -buildvcs=false ./tools/bootnode
+RUN apt-get install patch
+RUN bazelisk build //cmd/beacon-chain //cmd/validator
 
 FROM ubuntu:20.04
 RUN apt-get update && apt-get install -y curl jq iproute2
 
-COPY --from=builder /build/beacon-node /usr/local/bin/
-COPY --from=builder /build/validator /usr/local/bin/
+COPY --from=builder /app/prysm/bazel-out/*-fastbuild/bin/cmd/beacon-chain/beacon-chain_/beacon-chain /usr/local/bin/
```
Comment (Inphi) [inline prysm/Dockerfile.prysm]: I don't think globbing works with the COPY command.

Comment (0x00101010) [reply prysm/Dockerfile.prysm]: It is supported actually: https://docs.docker.com/engine/reference/builder/#copy

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -28,24 +28,28 @@ jobs:
         # Ignore the failure of a step and avoid terminating the job.
         continue-on-error: true
         with:
-          key: docker-cache-${{ hashFiles('.git/modules/prysm/*/HEAD', '.git/modules/geth/*/HEAD') }}
+          key: docker-cache-${{ hashFiles('.git/modules/prysm/*/HEAD', '.git/modules/geth/*/HEAD', './shared/Dockerfile.shared') }}
+
+      # build images as a separate step to better utilize caching
+      # TODO: change to pure "docker-compose build" once we can successfully build lighthouse and lodster
+      - run: docker-compose build prysm-beacon-node prysm-beacon-node-follower prysm-validator-node build-shared-states execution-node execution-node-2
```
Comment (0x00101010) [inline .github/workflows/ci.yml:35]: This saves lots of time by caching containers, each of the following tests only took 5 min to run now

File: .gitmodules [hunk 6]
```diff
@@ -4,8 +4,12 @@
...
 [submodule "prysm/prysm"]
 	path = prysm/prysm
-	url = https://github.com/Inphi/prysm
-	branch = eip-4844
+	; current branch contains latest commit from https://github.com/prysmaticlabs/prysm
+	; with applied fixes (to be merged) to support go build
+	; TODO: change back to original repo once the fix is merged
+	url = https://github.com/0x00101010/prysm
+	; url = https://github.com/prysmaticlabs/prysm
```
Comment (Inphi) [inline .gitmodules:11]: is there a PR that upstreams the fix?

File: docker-compose.yml [hunk 7]
```diff
@@ -135,29 +129,14 @@ services:
...
-      --grpc-gateway-port 7500
-      --interop-num-validators=4
-      --interop-start-index=0
-      --force-clear-db
-      --chain-config-file=/config/chain-config.yml
-      --suggested-fee-recipient 0xa94f5374fce5edbc8e2a8697c15331677e6ebf0b
-      --enable-tracing
-      --tracing-endpoint http://jaeger-tracing:14268/api/traces
-      --tracing-process-name validator-node
+      validator --accept-terms-of-use --beacon-rpc-provider prysm-beacon-node:4000 --rpc --grpc-gateway-host 0.0.0.0 --grpc-gateway-port 7500 --interop-num-validators=4 --interop-start-index=0 --force-clear-db --chain-config-file=/shared/chain-config.yml --suggested-fee-recipient 0xa94f5374fce5edbc8e2a8697c15331677e6ebf0b --enable-tracing --tracing-endpoint http://jaeger-tracing:14268/api/traces --tracing-process-name validator-node
```
Comment (Inphi) [inline docker-compose.yml]: nit: can we have this on multiple lines like before. It's harder to read.

File: prysm/run_beacon_node_peer.sh [hunk 8]
```diff
@@ -30,4 +30,4 @@ fi
 run_beacon_node.sh \
     --min-sync-peers=1 \
     --bootstrap-node "$PEER" \
-    --interop-genesis-time "$INTEROP_GENESIS_TIME" $@
+    --interop-num-validators 4 $@
```
Comment (Inphi) [inline prysm/run_beacon_node_peer.sh]: I could be wrong, but shouldn't the main beacon node be the one generating validators? If only the peer does this then we cant run validators without the peer.

File: shared/Dockerfile.shared [hunk 9]
```diff
@@ -0,0 +1,25 @@
+FROM golang:1.18 as builder
```
Comment (Inphi) [inline shared/Dockerfile.shared]: I dont think this Dockerfile is necessary. You can reuse the prysm one while changing the entrypoint for me the build-shared-states service.

**Review by Inphi — COMMENTED** [review]
Looks ok overall.

File: .gitmodules (hunk 6 above)
Comment (0x00101010) [reply .gitmodules:11]: I'll submit one later today

File: docker-compose.yml (hunk 7 above)
Comment (0x00101010) [reply docker-compose.yml]: np, this is auto fomratted by VS code, I'll change it back

File: shared/Dockerfile.shared (hunk 9 above)
Comment (0x00101010) [reply shared/Dockerfile.shared]: good catch, changed to use `Dockerfile.prysm`

File: .gitmodules (hunk 6 above)
Comment (0x00101010) [reply .gitmodules:11]: https://github.com/prysmaticlabs/prysm/pull/11901 here it is

**Review by Inphi — APPROVED** [review]
LGTM! Nice work

**Pull request merged**